crates/re_types_builder/src/objects.rs

```diff
@@ -0,0 +1,826 @@
+                    .filter(|utype| utype.base_type() != FbsBaseType::None)
+                    .is_some()
+            })
+            .map(|val| ObjectField::from_raw_enum_value(enums, objs, enm, &val))
+            .collect();
+
+        Self {
+            filepath,
+            fqname,
+            pkg_name,
+            name,
+            docs,
+            kind,
+            attrs,
+            fields,
+            specifics: ObjectSpecifics::Union { utype },
+        }
+    }
+
+    pub fn get_attr<T>(&self, name: impl AsRef<str>) -> T
+    where
+        T: std::str::FromStr,
+        T::Err: std::error::Error + Send + Sync + 'static,
+    {
+        self.attrs.get(self.fqname.as_str(), name)
+    }
+
+    pub fn try_get_attr<T>(&self, name: impl AsRef<str>) -> Option<T>
+    where
+        T: std::str::FromStr,
+        T::Err: std::error::Error + Send + Sync + 'static,
+    {
+        self.attrs.try_get(self.fqname.as_str(), name)
+    }
+
+    /// Returns the mandatory `order` attribute of this object.
+    ///
+    /// Panics if no order has been set.
+    pub fn order(&self) -> u32 {
+        self.attrs.get::<u32>(&self.fqname, "order")
+    }
+
+    pub fn is_struct(&self) -> bool {
+        match &self.specifics {
+            ObjectSpecifics::Struct {} => true,
+            ObjectSpecifics::Union { utype: _ } => false,
+        }
+    }
+
+    pub fn is_enum(&self) -> bool {
+        match &self.specifics {
+            ObjectSpecifics::Struct {} => false,
+            ObjectSpecifics::Union { utype } => utype.is_some(),
+        }
+    }
+
+    pub fn is_union(&self) -> bool {
+        match &self.specifics {
+            ObjectSpecifics::Struct {} => false,
+            ObjectSpecifics::Union { utype } => utype.is_none(),
+        }
+    }
+}
+
+/// Properties specific to either structs or unions, but not both.
+#[derive(Debug, Clone)]
+pub enum ObjectSpecifics {
+    Struct {},
+    Union {
+        /// The underlying type of the union.
+        ///
+        /// `None` if this is a union, some value if this is an enum.
+        utype: Option<ElementType>,
+    },
+}
+
+/// A high-level representation of a flatbuffers field, which can be either a struct member or a
+/// union value.
+#[derive(Debug, Clone)]
+pub struct ObjectField {
+    /// File path of the associated fbs definition, e.g. `//rerun/components/point2d.fbs`.
+    pub filepath: String,
+
+    /// Fully-qualified name of the field, e.g. `rerun.components.Point2D#position`.
+    pub fqname: String,
+
+    /// Fully-qualified package name of the field, e.g. `rerun.components`.
+    pub pkg_name: String,
+
+    /// Name of the object, e.g. `Point2D`.
+    pub name: String,
+
+    /// The field's multiple layers of documentation.
+    pub docs: Docs,
+
+    /// The field's type.
+    pub typ: Type,
+
+    /// The field's attributes.
+    pub attrs: Attributes,
+
+    /// Whether the field is required.
+    ///
+    /// Always true for `struct` types.
```

Comment:
saying `struct` here threw me off a bit at first - this is about flatbuffer structs, not about our higher level Struct vs Union interpretation, right? Maybe clarify somehow